src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -87,6 +87,41 @@
     }, "Configuration for the CephFS share")
 }
 
+JOIN_AUTH_SCHEMA = {
+    "resource_type": (str, "ceph.smb.join.auth"),
+    "auth_id": (str, "Unique identifier for the join auth resource"),
+    "intent": (str, "Desired state of the resource, e.g., 'present' or 'removed'"),
+    "auth": ({
+        "username": (str, "Username for authentication"),
+        "password": (str, "Password for authentication")
+    }, "Authentication credentials"),
+    "linked_to_cluster": (str, "Optional string containing a cluster ID. \
+    If set, the resource is linked to the cluster and will be automatically removed \
+    when the cluster is removed")
+}
+
+LIST_JOIN_AUTH_SCHEMA = [JOIN_AUTH_SCHEMA]
+
+USERSGROUPS_SCHEMA = {
+    "resource_type": (str, "ceph.smb.usersgroups"),
+    "users_groups_id": (str, "A short string identifying the usersgroups resource"),
+    "intent": (str, "Desired state of the resource, e.g., 'present' or 'removed'"),
+    "values": ({
+        "users": ([{
+            "name": (str, "The user name"),
+            "password": (str, "The password for the user")
+        }], "List of user objects, each containing a name and password"),
+        "groups": ([{
+            "name": (str, "The name of the group")
+        }], "List of group objects, each containing a name")
+    }, "Required object containing users and groups information"),
+    "linked_to_cluster": (str, "Optional string containing a cluster ID. \
+    If set, the resource is linked to the cluster and will be automatically removed \
+    when the cluster is removed")
+}
+
+LIST_USERSGROUPS_SCHEMA = [USERSGROUPS_SCHEMA]
+
```

Comment:
schema keeps on growing ;)

Reply:
Yeah... I realized I can at least generate other needed schemas from just a main one for each resource, so I'll be doing that.
I have it in another commit https://github.com/ceph/ceph/pull/61315/files#diff-293f6bd157fd219cf75ef48e0d571752e8bee7354bfd38921f2733c6e1f55426R96